### PR TITLE
Enable annotations for sdk service account

### PR DESCRIPTION
**Pull request by sudermanjr**
Signed-off-by: Andrew Suderman <andrew@sudermanjr.com>

**What type of PR is this?**
> Uncomment only one ` /kind <>` line, press enter to put that in a new line, and remove leading whitespace from that line:
>

/kind feature

**What this PR does / Why we need it**:
Adds the ability to annotate the sdk service account. Useful if you need to use workload identity on your gameservers.

### CODE REVIEW

**Comment by markmandel** [comment]
/cc @roberthbailey 

Just for my own edification - #2134 didn't cover enough for this I take it?

**Comment by markmandel** [comment]
```
Makefile:484: recipe for target 'gen-install' failed
make[1]: Leaving directory '/workspace/build'
Error: parse error at (agones/templates/serviceaccounts/sdk.yaml:73): unexpected EOF
```

😞 

To repro this issue locally (and you may need to run this anyway if it creates changes in index.yaml) -- run `make gen-install` to generate the index.yaml.

File: install/helm/agones/templates/serviceaccounts/sdk.yaml [hunk 1]
```diff
@@ -23,6 +23,9 @@ metadata:
     chart: {{ template "agones.chart" $ }}
     release: {{ $.Release.Name }}
     heritage: {{ $.Release.Service }}
+{{- if .Values.agones.serviceaccount.sdk.annotations }}
```
Comment (roberthbailey) [inline install/helm/agones/templates/serviceaccounts/sdk.yaml:26]: This if needs a matching `{{- end }}`.

Comment (roberthbailey) [reply install/helm/agones/templates/serviceaccounts/sdk.yaml:26]: From the build error: `Error: parse error at (agones/templates/serviceaccounts/sdk.yaml:73): unexpected EOF`

File: site/content/en/docs/Installation/Install Agones/helm.md [hunk 2]
```diff
@@ -190,6 +190,7 @@ The following tables lists the configurable parameters of the Agones chart and t
 | Parameter                                           | Description                                                                                     | Default                |
 | --------------------------------------------------- | ----------------------------------------------------------------------------------------------- | ---------------------- |
 | `agones.allocator.disableSecretCreation`            | Disables the creation of any allocator secrets. If true, you MUST provide the `allocator-tls`, `allocator-tls-ca`, and `allocator-client-ca` secrets before installation. | `false` |
+| `agones.serviceaccount.sdk.annotations`             | Annotations to add to the sdk service account in each gameserver namespace. | `{}` |
```
Comment (roberthbailey) [inline site/content/en/docs/Installation/Install Agones/helm.md:193]: nit: Can you write "[Annotations][annotations]" so it turns into a link like in the other rows?

**Review by roberthbailey — APPROVED** [review]

**Comment by sudermanjr** [comment]
> Just for my own edification - #2134 didn't cover enough for this I take it?

@markmandel  - I technically could have brought my own service account with annotations, but this seemed easier for this particular use.

File: install/helm/agones/templates/serviceaccounts/sdk.yaml (hunk 1 above)
Comment (sudermanjr) [reply install/helm/agones/templates/serviceaccounts/sdk.yaml:26]: Thanks for finding that.

**Comment by markmandel** [comment]
@sudermanjr 👋🏻 since you are on an your org's repo, I can't edit the PR. This has been approved, but I can't bring it up to date 😃 so if you could resolve the conflicts and rebase up to `main` we can get this merged 👍🏻

**Comment by roberthbailey** [comment]
@sudermanjr said in slack: "just realized i'm in the middle of a range statement there. I'll have to tweak this a lot more" so it needs more than just a rebase before the tests will pass.

**Comment by markmandel** [comment]
> @sudermanjr said in slack: "just realized i'm in the middle of a range statement there. I'll have to tweak this a lot more" so it needs more than just a rebase before the tests will pass.

Oh that's right! I forgot!

**Comment by markmandel** [comment]
Gently bumping this PR - just wanted to see if you plan on finishing this off? Marking as stale for the moment, since it's been a couple of weeks. Will close end of next week if we don't hear otherwise.

**Comment by markmandel** [comment]
Closing this PR to clean things up. Please feel free to reopen if you want to return to it.

**Pull request closed without merging**